lib/Zonemaster/Backend/DB.pm

```diff
@@ -390,6 +390,9 @@ sub get_batch_job_result {
         }
     }
 
+    die Zonemaster::Backend::Error::ResourceNotFound->new( message => "Unknown batch", data => { batch_id => $batch_id } )
+        if ( $result{nb_running} == 0 and $result{nb_finished} == 0 );
+
```

Comment:
Why does not the code look for undefined batch ID instead? That would be easier to understand in the future.

Reply:
I was miser on database connection. I'll add a look into the DB.

Reply:
I updated the code.

Reply:
Nice.